### ISSUE TITLE
Workspace-level enabled extensions

### ISSUE THREAD

**Issue by d8vjork** [Microsoft/vscode#69958]
Hi there,

I was wondering if there's a place where I can manage the extensions enabled on the Workspace level (instead of the general), if so will be brilliant if we can have **a collapsable block like the Disabled/Enabled/Recommended on the extensions panel**, something like Workspace.

Also now is a little bit frustrating the way that this feature is organised, I always do something like _Disable_ and then re-enable the extension just for the workspace, the _Enable (Workspace)_ option.

**Comment by d8vjork** [comment]
This will be also related to this other #68527

**Comment by alexweininger** [comment]
I agree with @d8vjork. There must be an easier way to manage extensions on a per workspace level.

**Comment by sandy081** [comment]
Related - https://github.com/Microsoft/vscode/issues/68527

**Comment by szhu** [comment]
Hello @d8vjork, I had some similar thoughts. I'm going to reproduce [my comment](https://github.com/microsoft/vscode/issues/85735#issuecomment-584993761) below since it seems pretty similar to your original post -- wanted to figure out if we're talking about similar things!

> It would be great for [workspace recommended extensions](https://code.visualstudio.com/docs/editor/extension-gallery#_workspace-recommended-extensions) to install to the current workspace by default. This can allow developers to confidently install recommended extensions knowing that the extensions won't clutter up their editing experience working on other projects.
> 
> In addition, making it easier to install an extension for just a single workspace makes it easier for new developers to "delete everything and try again" if they get an unexpected UI rather abandon VSCode completely. This might be similar in philosophy to why npm installs node_modules within the folder by default.
> 
> In general, it would be great to have a better extensions management UI. I've onboarded many people onto VSCode (because I love VSCode, so thank you maintainers) but unfortunately many of them don't understand how extensions work until I've showed them the install flow a few times. Redesigning the entire experience might be a major undertaking, but hopefully it's fairly easy to optimize the flow of a developer who only _consumes_ extensions rather than one who shops around and plays around with many extensions.

(I found this thread because I was also redirected to #68527 -- thanks @sandy081 for linking me!)

**Comment by d8vjork** [comment]
@szhu My thing is actually about the enabled ones.

So in your comment you describe the `.vscode/extensions.json` file, I'm actually using it in all of my projects where I'm working remotely with a team (and everyone use VS Code).

So as I can see, you've the same final conclusion, when I was having this problem is when I discovered that extensions file, but we still need something about that "Workspace extensions" / "User extensions", so teams can work with their own extensions and in the workspace we save in .vscode folder the preferences about which ones we enable (kind of like the Settings are doing right now).

Is very difficult to work in so many projects as in an agency and keep changing those new extensions you installed in other projects (they're installed and enabled as global / user extensions).

**Comment by blindjustis** [comment]
Hi - Just to add a vote for being able to configure extensions for a workspace.
Possibly even downloading them there rather than globally.

Kind of like npm for visual studio code.

The current recommendations feature is kinda useful for teams, but if you work on lots of different environments/languages it doesn't really fullfill the requirement.

**Comment by karlhorky** [comment]
This would be nice, yes. For example, when you would like to disable or enable certain extensions per-workspace if they conflict with each other.

Eg [`vscode-styled-jsx` conflicts with `vscode-styled-components`](https://github.com/iFwu/vscode-styled-jsx/issues/4#issuecomment-646619998)

**Comment by joeynguyen** [comment]
Correct me if I'm wrong, but this sounds to me like the same feature request as this issue? https://github.com/microsoft/vscode/issues/69958.  Just wanted to mention that so that when one is closed, the other will get closed as well.

I'm throwing in my vote for this feature request as well.  Currently I'm testing out using Remote Containers (https://code.visualstudio.com/docs/remote/containers#_managing-extensions) to accomplish this but it's overkill and uses additional resources for something like this.

**Comment by joeynguyen** [comment]
Correct me if I'm wrong, but this sounds to me like the same feature request as this issue? https://github.com/microsoft/vscode/issues/85735.  Just wanted to mention that so that when one is closed, the other will get closed as well.

I'm throwing in my vote for this feature request as well.  Currently I'm testing out using Remote Containers (https://code.visualstudio.com/docs/remote/containers#_managing-extensions) to accomplish this but it's overkill and uses additional resources for something like this.
